File: DoitMission_10/Doit10_Java/app/src/main/java/com/example/doit10_java/MyPagerAdapter.java
package com.example.doit10_java;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentPagerAdapter;

import java.util.ArrayList;

public class MyPagerAdapter extends FragmentPagerAdapter {

    private EmailFragment emailFragment = new EmailFragment();
    private InfoFragment infoFragment = new InfoFragment();
    private MapFragment mapFragment = new MapFragment();
    private int tabCount = 0;

    public MyPagerAdapter(FragmentManager fm, int tabCount){
        super(fm, BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT);
        this.tabCount = tabCount;
    }

    @Override
    public int getCount() {
        return tabCount;
    }

    @NonNull
    @Override
    public Fragment getItem(int position) {
        switch (position){
            case 0:
                return emailFragment;
            case 1:
                return infoFragment;
            case 2:
                return mapFragment;
                default:
                    return null;
        }
    }
}

File: DoitMission_10/Doit10_Kotlin/app/src/main/java/com/example/doit10_kotlin/MyPagerAdapter.kt
package com.example.doit10_kotlin

import android.util.Log
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter

class MyPagerAdapter: FragmentPagerAdapter {

    var emailFragment: EmailFragment = EmailFragment()
    var infoFragment: InfoFragment = InfoFragment()
    var locationFragment: LocationFragment = LocationFragment()
    var tabCount:Int = 0

    constructor(fm : FragmentManager, count:Int) : super(fm, BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT) {
        tabCount = count
        Log.e("Start", " 여기 들어옴 $tabCount")
    }


    override fun getItem(position: Int): Fragment {
        when(position){
            0 ->    return emailFragment
            1 ->    return infoFragment
            2 ->    return locationFragment
        }
        return emailFragment
    }

    override fun getCount(): Int {
        return tabCount
    }
}